R/epichains.R

```diff
@@ -55,19 +55,21 @@ new_epichains <- function(tree_df,
 #' @author James M. Azam
 #' @export
 epichains <- function(tree_df,
-                           index_cases,
-                           statistic,
-                           offspring_dist,
-                           stat_max,
-                           track_pop) {
+                      index_cases,
+                      statistic,
+                      offspring_dist,
+                      track_pop,
+                      stat_max = Inf) {
   # Check that inputs are well specified
-  checkmate::assert_data_frame(tree_df)
-  checkmate::assert_integerish(index_cases, null.ok = TRUE)
-  checkmate::assert_character(statistic, null.ok = TRUE)
+  checkmate::assert_data_frame(tree_df, min.cols = 3, min.rows = index_cases)
+  checkmate::assert_count(index_cases, positive = TRUE)
```

Comment:
Should this value have an upper bound that is a function of `tree_df`? I see that `tree_df` has a list of infectors - does the number of unique infectors limit the number of index cases? I don't understand the theory very well, so might be a naive or stupid question.

Reply:
> I see that `tree_df` has a list of infectors - does the number of unique infectors limit the number of index cases? I don't understand the theory very well, so might be a naive or stupid question.

No, you're right. The upper bound of `index_cases` should be the max of the `infectee_id` in `tree_df`. I've added the check in 236306d7d991b2dc0e8eff15f2775e9c610e3eb6.